src/chatdDb.h

```diff
@@ -604,6 +612,32 @@ class ChatdSqliteDb: public chatd::DbInterface
             reactions.insert(std::pair<std::string, karere::Id>(stmt.stringCol(0), stmt.uint64Col(1)));
         }
     }
+
+    void getIdxByRetentionTime(const time_t ts, chatd::Idx &limitIdx) override
+    {
+        // Find the most recent msg affected by retention time if any
+        SqliteStmt stmt(mDb, "select MAX(ts), MAX(idx) from history where chatid = ? and ts < ?");
+        stmt.bind(mChat.chatId())
+                .bind(ts)
+                .step();
+
+        if (sqlite3_column_type(stmt, 0) != SQLITE_NULL)
+        {
+            limitIdx = stmt.intCol(1);
+        }
+        else
+        {
+            limitIdx = CHATD_IDX_INVALID;
+        }
+    }
```

Comment:
```suggestion
    chatd::Idx getIdxByRetentionTime(const time_t ts) override
    {
        // Find the most recent msg affected by retention time, if any
        SqliteStmt stmt(mDb, "select MAX(ts), MAX(idx) from history where chatid = ? and ts < ?");
        stmt << mChat.chatId() << ts;
        return (stmt.step()) ? stmt.intCol(1) : CHATD_IDX_INVALID;
    }
```

Reply:
This query always will have a step, so we need the following condition to differentiate if query returned a valid row or NULL:
(sqlite3_column_type(stmt, 0) != SQLITE_NULL)

Reply:
IC. You use the `max()` and that's why the step success (queries ALWAYS have an step and it may result on a `SQLITE_ROW` or on a `SQLITE_DONE`). What you mean is that in this case, even when there's no `MAX()` values, it still returns `SQLITE_ROW`, but the values are `SQLITE_NULL`. 
Your code, however, checks for `MAX(ts)` not being null, and then it reads the `MAX(idx)`, so it may be 0, which will result on unexpected behavior, since that index corresponds to a different message.

Reply:
```suggestion
    void getIdxByRetentionTime(const time_t ts, chatd::Idx &limitIdx) override
    {
        // Find the most recent msg affected by retention time if any
        SqliteStmt stmt(mDb, "select MAX(ts), MAX(idx) from history where chatid = ? and ts < ?");
        stmt << mChat.chatId() << ts;
        return (stmt.step() && sqlite3_column_type(stmt, 1) != SQLITE_NULL) ? stmt.intCol(1) : CHATD_IDX_INVALID;
    }
```